### ISSUE TITLE
Sticky: Poll: What do you want to see in Zenbot 3.6?

### ISSUE THREAD

**Issue by carlos8f** [DeviaVir/zenbot#41]
Greetings Zenbot users!

After a gazillion commits over the weekend Zenbot 3.5.x is fairly stable now. However there's much more Zenbot is capable of, and only so much time to implement it. I need help prioritizing!
- [x] Auto-learn engine, to mutate trade params on the fly, simulate in parallel, and hot-reload better params into the runner?
- [x] Simulation with date ranges?
- [x] Generic `default_trade.js` usable with any exchange?
- [x] Composable JSON configuration?
- [x] More exchanges?
- [x] Better support for EUR trading and other currencies
- [x] Web UI for bot status or control?
- [x] Email or text message on trade execution?
- [x] New indicators such as MACD?
- [x] Repair tool for backfilled data
- [ ] Improve the ROI for BTC?
- [ ] Improve the ROI for ETH?
- [ ] More altcoins support?
- [ ] Twitter integration?
- [ ] Performance?
- [ ] Documentation or guides?
- [ ] How-to videos?
- [ ] More "hint" or "help" notifications when executing commands?
- [ ] man page?
- [ ] Online docs for Zenbrain engine and API?
- [ ] Online docs for writing plugins?
- [ ] Example apps using Zenbrain for other purposes? (I have a web crawler and twitter bot built on earlier versions of Zenbrain)
- [ ] Backup or maintenance scripts?
- [ ] Additional commands? Utilities?
- [ ] Dockerfile?

Please comment below with what features you'd like to see in the next release, 3.6.0.

Cheers,
Carlos

**Comment by grigio** [comment]
- Generic default_trade.js (Exchange indipendent logic/strategy, abstract buy, sell, order_check)

**Comment by grigio** [comment]
- Simulation with data ranges #40

**Comment by grigio** [comment]
- Composable JSON configuration https://github.com/carlos8f/zenbot/issues/36#issuecomment-241827313

**Comment by itsgarryb** [comment]
- Auto-learn engine, to mutate trade params on the fly, simulate in parallel, and hot-reload better params into the runner

If it works properly, this would be great to have.

**Comment by xangma** [comment]
Definitely this:
- Auto-learn engine, to mutate trade params on the fly, simulate in parallel, and hot-reload better params into the runner?

I know this is a bit of specific request, but if I wanted to run this on multiple markets on the same exchange, I would set the trade amount to: default trade amount/number of markets. I.e. 3 markets, trade 0.3 of my total each time. This would be to make sure there is enough money in the bank for every market. But this would only work if there were no double buy/sell signals on any single market. This is something I think deserves looking into.

More markets, more trading opportunities, more potential profit?

**Comment by tehGoti** [comment]
- More exchanges?

Especially support for EUR trades
- Web UI for bot status or control?

**Comment by BarnumD** [comment]
New indicators such as MACD..

**Comment by BarnumD** [comment]
Email Text notifications

**Comment by backenders-game** [comment]
Generic default_trade.js usable with any exchange
Guides/documentation how to use it with exchanges other than gdax.

Guides/documentation how to develop and tweak own strategies

**Comment by JeffUberman** [comment]
This may be a 4.0 or beyond, but I'd like to see 'smart portfolio diversification' across various cryptocurrencies (say, the 5 or 10 with the highest market cap or trading volume) and to intelligently shift portions of --but not entire- allocations to currencies that are deemed more favorable at present, making zenbot more tactical. For example, maybe current portfolio is 55% BTC, 25% ETH, 10% LTC, 10% mix of others...then some news breaks and BTC is on the downtrend (either more risky than an alternative or less of an expected ROI) so zenbot shifts allocation to 30% BTC, 45% ETC, 20% LTC, 5% other.

**Comment by carlos8f** [comment]
@JeffUberman Zenbot is already capable of that, I am currently using Zenbot 3.5.13 to trade ETH-BTC-USD on GDAX which is similar to what you're describing. It will buy/sell ETH or BTC depending on which is trending at the moment.

**Comment by totomaze** [comment]
Maybe a repair mode to find gaps in the data (missing data visible in the graph). I may be the only one to see this problem as I am running zenbot on an atom processor with 2gb of ram. The backfill seems ok but the reducer seems to skip some data.

**Comment by JFD3D** [comment]
A function for monitoring the top 5 or so highest 24 hour volume pairs on any given exchange would be nice :)

What @JeffUberman said is interesting. I haven't had a chance to test this yet so I will just ask, does zenbot currently trade multiple pairs and switch to another alt if for example you are running two scripts, ETH-BTC and LTC-BTC, the ETH-BTC pair is holding BTC and suddenly LTC trend triggers rs.op 'buy' will it will automatically switch to that pair (as in buy LTC)?

It would be awesome to have functionality that would predict pair profitability based on past performance and current trend data and switch to a more profitable currency while taking into account a 24 hour balance (and orderbooks?) to total user balance ratio. If the profitability margin is high enough maybe zenbot could even sell an asset where it stands and immediately buy into the more profitable opportunity. Also triangular arbitrage is fun although it probably lies outside the realm of this project.

In my opinion the more features/strategies/options parameters there are the less likely zenbot will suffer from "dilution syndrome" and lose it's profitability due to everyone using it although this may be more unlikely thanks to zenbrain. 

Keep up the good work!

**Comment by carlos8f** [comment]
@JFD3D yes, in that scenario the bot would buy LTC, as long as the pairs are traded on the same exchange and share the same account balance.

I agree it would be awesome if Zenbot could analyze the entire crypto portfolio for the most profitable things to focus on, this is out of scope for Zenbot 3 though, perhaps in 4 :)

**Comment by buzztiaan** [comment]
bittrex support ^_^

**Comment by lyn168** [comment]
More exchanges: bitstamp, bitfinex, CEXio

**Comment by wupeng1211** [comment]
More exchanges: btcc

**Comment by jltrade** [comment]
bollinger bands please!

**Comment by totomaze** [comment]
Have a option to prevent buy after buy or sell after sell.

**Comment by hlboys** [comment]
no error
no bug
Programs that can be used

**Comment by michaldev** [comment]
More exchanges or better support for EUR :)
Coinbase exchange isn't anonymous. They follow transactions, check id etc.
In coinbase person outside the USA can not trade USD. (Only EUR)

**Comment by carlos8f** [comment]
bump

**Comment by carlos8f** [comment]
![mila](https://cloud.githubusercontent.com/assets/106763/19346026/265a072c-90f6-11e6-8d39-d6d7c698256f.png)

**Comment by D4rk4** [comment]
Please add BTC-e.com exchange to zenbot.

**Comment by JFD3D** [comment]
@carlos8f Yes, pls Mila Kunis in 3.6 :P

**Comment by totomaze** [comment]
Set the start balance by Asset or by Currency.
or 
Set the first action of the Simulation (start with buy only).

**Comment by wiscokidd** [comment]
an option to place limit orders instead or market orders to avoid fees.

**Comment by coinchirp** [comment]
"an option to place limit orders instead or market orders to avoid fees" this 100x over, i was really surprised having only worked with/on HFT bots that this was not an option. Market fees always seem to kill bots over time and this logic applied could save some money, and help increase trading volume more reasonably

**Comment by kingo55** [comment]
Poloniex too please!

**Comment by carlos8f** [comment]
Alas, 3.6 never came, but Zenbot 4 is out, so I'm cleaning up 3.x issues. Please use and test 4, thanks!
